src/plugins/intel_cpu/src/compiled_model.cpp

```diff
@@ -90,7 +90,7 @@ CompiledModel::CompiledModel(const std::shared_ptr<ov::Model>& model,
 #if defined(OV_CPU_WITH_ACL)
                     static std::once_flag flag_once;
                     std::call_once(flag_once, [&]() {
-                        std::shared_ptr<arm_compute::IScheduler> acl_scheduler = std::make_shared<ACLScheduler>();
+                        std::shared_ptr<arm_compute::IScheduler> acl_scheduler = std::make_shared<ACLScheduler>(m_cfg.streamExecutorConfig.get_threads());
```

Comment:
How it should work if the application compiles 2 models in parallel with different number of threads? 
From my understanding still single ACLScheduler object will be created with the state (numebr of threads) which is not valid for one of the models.

Reply:
we may replace `std::min(num_iterations, m_threads)` with `num_iterations`, so we. won't call `parallel_get_num_threads()` at all in ACLScheduler.
`ov::parallel_for` finds min(work, threads) in TBB case, so TBB case shouldn't be affected.
If you don't mind I can do perf check of this change on TBB and OpenMP.

Reply:
Such change will affect the following branch in TBB case: 
![image](https://github.com/user-attachments/assets/b89b4338-ff72-472d-8c1a-6128f7d3ad28)
However we still can have a try and measure the perf. 
Alternativelly lets have a little bit different impls for TBB and OMP

Reply:
> Alternativelly lets have a little bit different impls for TBB and OMP

Do you suggest just to have 2 different `_num_threads` initialization wrapped by `#if OV_THREAD == OV_THREAD_TBB ... #elif OV_THREAD == OV_THREAD_OMP` to avoid any changes in TBB case?

Reply:
Yes, the main idea is to keep TBB path unchanged

Reply:
Applied this approach.

Reply:
Looks good. Lets add comment that will describe why we need different behavior for OMP